### PR TITLE
preserve data in the browser

### PR DESCRIPTION
**Pull request by den-cloud**
For this final PR for the portfolio website, I will save the form data in the local storage of the browser. That way when the user reloads the page the data they filled out in the form will be preserved.

### CODE REVIEW

File: portfolio.js [hunk 1]
```diff
@@ -127,3 +127,45 @@ form.addEventListener('submit', (e) => {
...
+      e.code === 22
+      // Firefox
+      || e.code === 1014
+      // test name field too, because code might not be present
+      // everything except Firefox
+      || e.name === 'QuotaExceededError'
+      // Firefox
+      || e.name === 'NS_ERROR_DOM_QUOTA_REACHED')
+      // acknowledge QuotaExceededError only if there's something already stored
+      && (storage && storage.length !== 0);
```
Comment (DanteAlonsoHT) [inline portfolio.js]: - [ ] [OPTIONAL] It will be nice if you remove all these comments make your code cleaner in your repo 😉

**Review by DanteAlonsoHT — CHANGES_REQUESTED** [review]
<h1> Status: Required Changes 🚫 </h1>

Hi @den-cloud, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>

## Required Changes ♻️

- [ ] - Follow the git workflow, we expect to see more commits if you want to have a reference (at least 5 commits for quick projects like this one, at least 8 or 10 for medium projects, and more than 15 or 20 for big projects) Keep up the good work! 

## Optional suggestions
_Check the comments below_

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- [ ] - Create "css" folder and put all .css files inside this folder.
- [ ] - Create "js" folder and put all .js files inside this folder.
- [ ] - Create "img" folder and put all .png, .jpeg, .jpg, .svg files (all images and vectors) inside this folder.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: portfolio.js (hunk 1 above)
Comment (den-cloud) [reply portfolio.js]: I totally confused on what you expect me to change. May you pleas help me

Comment (DanteAlonsoHT) [reply portfolio.js]: Sure! your project is working fine, just add more commits 😉 and as optional, you can remove the comments line in your JS code " // ".

Comment (den-cloud) [reply portfolio.js]: okay thanks

Comment (DanteAlonsoHT) [reply portfolio.js]: You're welcome, reminder: at least 5 commits (for this project) would be nice 💯

Comment (den-cloud) [reply portfolio.js]: Thanks a lot for the constructive review man.

**Review by SajjadAhmad14 — APPROVED** [review]
# Status: Approved :heavy_check_mark: :heavy_check_mark: :heavy_check_mark:

Hi Denzel Nkomo, 
## Congratulations 🎉 Thank you for implementing changes suggested previously 🙏

## Happy Coding 👏👏👏
#### Please leave any questions or comments in the PR if you have any doubts or unclarity!
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

File: portfolio.js (hunk 1 above)
Comment (DanteAlonsoHT) [reply portfolio.js]: You're welcome, nice work! 💪

**Pull request merged**